src/amp-story-player/amp-story-player-impl.js

```diff
@@ -1270,6 +1280,21 @@ export class AmpStoryPlayer {
     });
   }
 
+  /**
+   * Sends a message to the story to navigate delta pages.
+   * @param {number} delta
+   * @private
+   */
+  selectPage_(delta) {
+    const navigation = delta > 0 ? {'next': true} : {'previous': true};
+    const {iframeIdx} = this.stories_[this.currentIdx_];
+    this.messagingPromises_[iframeIdx].then((messaging) => {
+      for (let i = 0; i < Math.abs(delta); i++) {
+        messaging.sendRequest('selectPage', navigation);
+      }
```

Comment:
@Enriqe changing this to
```
messaging.sendRequest('selectPage', {'page': pageId});
```

And in AmpStory add:
```
onSelectPage_(data) {
    ...
    } else if (data['page']) {
        this.switchTo_(data['page']);
    }
}
``` 

is what can allow stories to go to the page by pageIds